lib/modules/datasource/docker/index.ts

```diff
@@ -702,12 +712,19 @@ export class DockerDatasource extends Datasource {
       | string
       | null = `${registryHost}/${dockerRepository}/tags/list?n=${limit}`;
     url = ensurePathPrefix(url, '/v2');
-    const headers = await getAuthHeaders(
-      this.http,
-      registryHost,
-      dockerRepository,
-      url
-    );
+    let headers = null;
+    try {
+      headers = await getAuthHeaders(
+        this.http,
+        registryHost,
+        dockerRepository,
+        url
+      );
+    } catch (err) /* istanbul ignore next */ {
```

Comment:
i don't think we should catch here, as that's a fatal error and we should break any further processing of the dep.

Reply:
I put this catch here to handle the thrown exception from `getAuthHeaders()`, should I move it to the end of `getAuthHeaders()`?

Reply:
I rolled back this `try catch` and `getTags()` will catch the `404Error` and handle it.